src/snowflake/cli/plugins/git/manager.py

```diff
@@ -18,3 +20,36 @@ def show_files(self, repo_path: str) -> SnowflakeCursor:
     def fetch(self, repo_name: str) -> SnowflakeCursor:
         query = f"alter git repository {repo_name} fetch"
         return self._execute_query(query)
+
+    def create(
+        self, repo_name: str, api_integration: str, url: str, secret: str
+    ) -> SnowflakeCursor:
+        query = (
+            f"create git repository {repo_name}"
+            f" api_integration = {api_integration}"
+            f" origin = '{url}'"
+        )
+        if secret is not None:
+            query += f" git_credentials = {secret}"
+        return self._execute_query(query)
+
+    def create_secret(self, name: str, username: str, password: str) -> SnowflakeCursor:
+        query = (
+            f"create secret {name}"
+            f" type = password"
+            f" username = '{username}'"
+            f" password = '{password}'"
+        )
+        return self._execute_query(query)
+
+    def create_api_integration(
```

Comment:
Move to SQLMixin as those are not specific to git repo

Reply:
Done. I renamed `create_secret -> create_password_secret` as different secret types have different arguments